### ISSUE TITLE
[FORMATTING] When using `:` in my T-SQL queries, the formatter doesn't work via CLI

### ISSUE THREAD

**Issue by fsuarez-RMS** [sql-formatter-org/sql-formatter#632]
S**Input data**

Which SQL and options did you provide as input?

config.json

```json
{
    "language": "transactsql",
    "tabWidth": 1,
    "keywordCase": "upper",
    "linesBetweenQueries": 3,
    "commaPosition": "places",
    "denseOperators": true,
    "expressionWidth": 50,
    "indentStyle": "tabularLeft",
    "tabulateAlias": true,
    "logicalOperatorNewline": "after",
    "newlineBeforeSemicolon": true,
    "useTabs": true,
    "paramTypes": {
        "named": [":"]
        },
    "params": ["John", "27"]
    }

```

```sql
DECLARE @Counter int;  
SET @Counter = 1;  
WHILE @Counter < 10  
BEGIN   
    SELECT @Counter  
    SET @Counter = @Counter + 1  
    IF @Counter = 4 GOTO Branch_One --Jumps to the first branch.  
    IF @Counter = 5 GOTO Branch_Two  --This will never execute.  
END  
Branch_One:  
    SELECT 'Jumping To Branch One.'  
    GOTO Branch_Three; --This will prevent Branch_Two from executing.  
Branch_Two:  
    SELECT 'Jumping To Branch Two.'  
Branch_Three:  
    SELECT 'Jumping To Branch Three.';
```

**Expected Output**

```sql
DECLARE @Counter INT;



SET
	@Counter=1;



WHILE @Counter<10 BEGIN
SELECT
	@Counter
SET
	@Counter=@Counter+1 IF @Counter=4 GOTO Branch_One --Jumps to the first branch.
	IF @Counter=5 GOTO Branch_Two --This will never execute.
	END Branch_One
SELECT
	'Jumping To Branch One.' GOTO Branch_Three;



--This will prevent Branch_Two from executing.
Branch_Two
SELECT
	'Jumping To Branch Two.' Branch_Three
SELECT
	'Jumping To Branch Three.';
```

**Actual Output**

```sql

Error: Parse error: Unexpected ":
    SELE" at line 10 column 11
    at iR.createParseError (C:\Users\fsuarez\AppData\Roaming\nvm\v18.17.1\node_modules\sql-formatter\dist\sql-formatter.min.cjs:1:191542)
    at iR.tokenize (C:\Users\fsuarez\AppData\Roaming\nvm\v18.17.1\node_modules\sql-formatter\dist\sql-formatter.min.cjs:1:191328)
    at YR.tokenize (C:\Users\fsuarez\AppData\Roaming\nvm\v18.17.1\node_modules\sql-formatter\dist\sql-formatter.min.cjs:1:192718)
    at JR.tokenize (C:\Users\fsuarez\AppData\Roaming\nvm\v18.17.1\node_modules\sql-formatter\dist\sql-formatter.min.cjs:1:225154)
    at JR.reset (C:\Users\fsuarez\AppData\Roaming\nvm\v18.17.1\node_modules\sql-formatter\dist\sql-formatter.min.cjs:1:197686)
    at T.I.feed (C:\Users\fsuarez\AppData\Roaming\nvm\v18.17.1\node_modules\sql-formatter\dist\sql-formatter.min.cjs:1:4085)
    at Object.parse (C:\Users\fsuarez\AppData\Roaming\nvm\v18.17.1\node_modules\sql-formatter\dist\sql-formatter.min.cjs:1:225309)
    at PA.parse (C:\Users\fsuarez\AppData\Roaming\nvm\v18.17.1\node_modules\sql-formatter\dist\sql-formatter.min.cjs:1:225504)
    at PA.format (C:\Users\fsuarez\AppData\Roaming\nvm\v18.17.1\node_modules\sql-formatter\dist\sql-formatter.min.cjs:1:225017)
    at rA (C:\Users\fsuarez\AppData\Roaming\nvm\v18.17.1\node_modules\sql-formatter\dist\sql-formatter.min.cjs:1:229805)

Node.js v18.17.1

Process finished with exit code 1
```

**Usage**

- How are you calling / using the library?   CLI installation via npm
- What SQL language(s) does this apply to? T-SQL
- Which SQL Formatter version are you using? 12.2.4

**Comment by nene** [comment]
Thanks for reporting. Unfortunately the formatter doesn't really support procedural SQL.

Though it would be nice if it didn't crash when encountering a goto-label. There's a slight hope I can patch this, but I wouldn't really bet on it getting fixed.

**Comment by nene** [comment]
I cobbled up a rudimentary fix for this, so at least it doesn't crash, though the formatting of the above code is still quite terrible:

```sql
DECLARE @Counter int;

SET
  @Counter = 1;

WHILE @Counter < 10 BEGIN
SELECT
  @Counter
SET
  @Counter = @Counter + 1 IF @Counter = 4 GOTO Branch_One --Jumps to the first branch.
  IF @Counter = 5 GOTO Branch_Two --This will never execute.
  END Branch_One:
SELECT
  'Jumping To Branch One.' GOTO Branch_Three;

--This will prevent Branch_Two from executing.
Branch_Two:
SELECT
  'Jumping To Branch Two.' Branch_Three:
SELECT
  'Jumping To Branch Three.';
```

**Comment by nene** [comment]
Closing this issue as the crash has been fixed, but the actual support for formatting procedural SQL is not planned.

**Comment by nene** [comment]
Published the fix in 13.0.0
